src/main.js

```diff
@@ -386,12 +395,17 @@ const CodingChatboxMainWindow = new Lang.Class({
         this._clockSettings = new Gio.Settings({ schema: CLOCK_SCHEMA });
         this._clockSettings.connect('changed::' + CLOCK_FORMAT_KEY,
                                     Lang.bind(this, this._updateClockFormat));
-        this.attachment_preview_close.connect('clicked', Lang.bind(this, function() {
-            this.attachment_preview_image.clear();
-            this.attachment_preview_filename.label = '';
-            this.attachment_preview_desc.label = '';
-            this.chatbox_view_stack.set_visible_child_name('chats');
-        }));
+        this.attachment_preview_close.connect('clicked',
+                                              Lang.bind(this, this._closeAttachmentPreview));
+        this.add_accel_group(createAccelGroupFor('Escape',
```

Comment:
Another advantage is that through `GtkActionable` you can also associate the same action to the button, without having to connect to the clicked event directly.